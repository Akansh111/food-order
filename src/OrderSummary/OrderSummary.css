.order-summary {
    width: 35%;
    margin: 24px 35px;
    border: 1px solid black;
    border-radius: 25px;
    padding: 20px;
    text-align: center;
}

.card-title {
    font-size: 32px;
    font-weight: bold;
}
.summary-table {
    height: 230px;
    overflow: auto;
}
.cart-total {
    font-size: 23px;
    font-weight: bold;
}