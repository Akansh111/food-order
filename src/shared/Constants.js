export const url = 'https://demo8151081.mockable.io/getFoodData';
export const mainTitle = 'Lets BANG ON to your Favourite Burger';
export const orderNowTitle = 'ORDER NOW!!!';
export const menuMessage = 'Select the items from the Menu!';
export const successMessage = 'Wohoo, Your order will be delivered in 30 minutes!';
export const discountMessage = 'after applying discount';
export const orderSummaryTitle = 'ORDER SUMMARY';