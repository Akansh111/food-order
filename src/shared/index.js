export * from './Constants';
export * from './mockApiData';
