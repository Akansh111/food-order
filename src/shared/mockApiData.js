export const mockApiData = [
    {
        "burgerName": "Panner King's Melt",
        "rating": "★★★★★",
        "price": 129,
        "category": "Veg",
        "discount": 10
    },
    {
        "burgerName": "Veggie",
        "rating": "★★★",
        "price": 95,
        "category": "Veg",
        "discount": 5
    },
    {
        "burgerName": "Makani Supreme Veg",
        "rating": "★★★★★",
        "price": 79,
        "category": "Veg",
        "discount": 5
    },
    {
        "burgerName": "Crispy Veg",
        "rating": "★★★★",
        "price": 55,
        "category": "Veg",
        "discount": 5
    },
    {
        "burgerName": "Fiery Chicken",
        "rating": "★★★★",
        "price": 159,
        "category": "Non-Veg",
        "discount": 15
    },
    {
        "burgerName": "Chicken Chilli Cheese Melt",
        "rating": "★★★★★",
        "price": 129,
        "category": "Non-Veg",
        "discount": 10
    },
    {
        "burgerName": "Crispy Chicken",
        "rating": "★★★",
        "price": 75,
        "category": "Non-Veg",
        "discount": 5
    },
    {
        "burgerName": "Crispy Chicken Supreme",
        "rating": "★★★★★",
        "price": 85,
        "category": "Non-Veg",
        "discount": 5
    }
]