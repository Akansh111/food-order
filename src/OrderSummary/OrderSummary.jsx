import React from 'react';
import { Button, Modal } from 'react-bootstrap';
import { menuMessage, successMessage, discountMessage, orderSummaryTitle } from '../shared';
import './OrderSummary.css';

export const OrderSummary = (props) => {
    const { summaryData } = props;
    const [show, setShow] = React.useState(false);
    let total = 0.00;
    let summaryJsx = summaryData && summaryData.map((item, index) => {
        const finalPrice = (item.price * (1 - item.discount/100)).toFixed(2);
        total = total + (item.price * (1 - item.discount/100));
        return (
            <tr key={index}>
                <td>
                    {item.burgerName} - {'₹' + item.price}
                </td>
                <td>{' = '}</td>
                <td>
                    <b>
                        {`₹ ${finalPrice} (${discountMessage})`}
                    </b>
                </td>
            </tr>
        );
    });

    const handleClose = () => {
        setShow(false);
        window.location.reload();
    }
    const handleShow = () => setShow(true);

    return (
        <div className="order-summary">
            <div className="card-title">
                {orderSummaryTitle}
            </div>
            <br />
            <div className="summary-table">
                <table className="table table-striped">
                    <tbody>
                        {summaryJsx.length ? summaryJsx : <div>{menuMessage}</div>}
                    </tbody>
                </table>
            </div>
            <div className="cart-total">
                {'Total: ₹' + total.toFixed(2)}
            </div>
            <Button variant="primary" size="lg" block onClick={handleShow} disabled={total === 0}>
                Confirm Order
            </Button>

            <Modal show={show} onHide={handleClose}>
                <Modal.Header closeButton>
                <Modal.Title>CONGRATULATIONS</Modal.Title>
                </Modal.Header>
                <Modal.Body>{successMessage}</Modal.Body>
                <Modal.Footer>
                <Button variant="secondary" onClick={handleClose}>
                    Close
                </Button>
                </Modal.Footer>
            </Modal>
        </div>
    )
}
