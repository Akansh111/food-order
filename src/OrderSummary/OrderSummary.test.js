import React from 'react';
import renderer from 'react-test-renderer';
import { mockApiData } from '../shared'
import { OrderSummary } from './OrderSummary';

describe('renders OrderSummary', () => {
    const mockProps = { "summaryData": [{ "burgerName": "Panner King's Melt", "rating": "★★★★★", "price": 129, "category": "Veg", "discount": 10 }] };
    it('renders', () => {
        const component = renderer.create(<OrderSummary { ...mockProps } />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});