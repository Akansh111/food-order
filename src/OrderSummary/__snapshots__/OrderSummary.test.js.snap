// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders OrderSummary renders 1`] = `
<div
  className="order-summary"
>
  <div
    className="card-title"
  >
    ORDER SUMMARY
  </div>
  <br />
  <div
    className="summary-table"
  >
    <table
      className="table table-striped"
    >
      <tbody>
        <tr>
          <td>
            Panner King's Melt
             - 
            ₹129
          </td>
          <td>
             = 
          </td>
          <td>
            <b>
              ₹ 116.10 (after applying discount)
            </b>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="cart-total"
  >
    Total: ₹116.10
  </div>
  <button
    className="btn btn-primary btn-block btn-lg"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Confirm Order
  </button>
</div>
`;
